src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1192,6 +1192,19 @@ void DataWriterImpl::InnerDataWriterListener::onWriterMatched(
         const PublicationMatchedStatus& info)
 {
     data_writer_->update_publication_matched_status(info);
+
+    StatusMask notify_status = StatusMask::publication_matched();
+    DataWriterListener* listener = data_writer_->get_listener_for(notify_status);
+    if (listener != nullptr)
+    {
+        PublicationMatchedStatus callback_status;
+        if (ReturnCode_t::RETCODE_OK == data_writer_->get_publication_matched_status(callback_status))
+        {
+            listener->on_publication_matched(data_writer_->user_datawriter_, callback_status);
+        }
+    } else {
```

Comment:
We should please the linters:

```suggestion
    }
    else
    {
```

Reply:
Updated